DuckDuckGo/TabsModelPersistenceExtension.swift

```diff
@@ -26,17 +28,34 @@ extension TabsModel {
     }
 
     public static func get() -> TabsModel? {
-        guard let data = UserDefaults.app.object(forKey: Constants.key) as? Data else { return nil }
-        return try? NSKeyedUnarchiver.unarchiveTopLevelObjectWithData(data) as? TabsModel
+        guard let data = UserDefaults.app.object(forKey: Constants.key) as? Data else {
+            return nil
+        }
+        var tabsModel: TabsModel?
+        do {
+            let unarchiver = try NSKeyedUnarchiver(forReadingFrom: data)
+            unarchiver.requiresSecureCoding = false
+            tabsModel = unarchiver.decodeObject(of: self, forKey: NSKeyedArchiveRootObjectKey)
+            if let error = unarchiver.error {
+                throw error
+            }
+        } catch {
+            os_log("Something went wrong unarchiving TabsModel %@", log: .generalLog, type: .error, error.localizedDescription)
+        }
+        return tabsModel
     }
 
     public static func clear() {
          UserDefaults.app.removeObject(forKey: Constants.key)
     }
 
     func save() {
-        guard let data = try? NSKeyedArchiver.archivedData(withRootObject: self, requiringSecureCoding: false) else { return }
-        UserDefaults.app.set(data, forKey: Constants.key)
+        var data: Data
```

Comment:
This is only used inside the block so might as well be defined there and can also be a `let`

Reply:
agree, fixed!